### PR TITLE
CronJob jsonschema validation

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/stormtypes.py [hunk 1]
```diff
@@ -2077,10 +2077,14 @@ def _parseAlias(self, opts):
...
+                'reqs': reqdict,
+                'incunit': incunit,
+                'incvals': incval,
+                'useriden': self.runt.user.iden
+                }
 
-        todo = s_common.todo('addCronJob', self.runt.user.iden, query, reqdict, incunit, incval)
+        todo = s_common.todo('addCronJob', cdef)
         gatekeys = ((self.runt.user.iden, ('cron', 'add'), None),)
         iden = await self.dyncall('cortex', todo, gatekeys=gatekeys)
```
Comment (invisig0th) [inline synapse/lib/stormtypes.py]: Lets make the addCronJob API return the new cdef ( to align with the other stormtypes uses )

File: synapse/lib/agenda.py [hunk 2]
```diff
@@ -528,14 +576,17 @@ def list(self):
         Returns:
             iden of new appointment
         '''
-        iden = croniden
+        iden = cdef['iden']
+        incunit = cdef.get('incunit')
+        incvals = cdef.get('incvals')
+        reqs = cdef.get('reqs', {})
+        query = cdef.get('storm')
+        useriden = cdef.get('useriden')
```
Comment (invisig0th) [inline synapse/lib/agenda.py]: Perhaps we should make this "creator" similar to the view / layer cdef?

File: synapse/cmds/cron.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ class Cron(s_cli.Cmd):
         Returns the iden that starts with prefix.  Prints out error and returns None if it doesn't match
         exactly one.
         '''
-        idens = [iden for iden, trig in await core.listCronJobs()]
+        idens = [cron.get('iden') for cron in await core.listCronJobs()]
```
Comment (jnwatson) [inline synapse/cmds/cron.py]: rather just `cron['iden']` as it will just blow up on next line if None.

Comment (jnwatson) [reply synapse/cmds/cron.py]: (And below)

**Pull request merged**